Engine/Module/ModuleResourceManager.h

```diff
@@ -147,6 +196,35 @@ class ModuleResourceManager : public Module
 		std::atomic_uint total_items = 0;
 	} thread_comunication;
 
+	//Asyncronous loading variables
+	int max_threads = 1;
+	std::atomic<bool> loading_threads_active = true;
+	std::vector<std::thread> loader_threads;
+	bool normal_loading_flag = false;
+	bool load_scene_asyncronously = true;
+	//We change this value to the component that calls Load for textures
+	Component* current_component_loading = nullptr;
+	ResourceType current_type = ResourceType::TEXTURE;
+	//Texture Type 
+	unsigned texture_type = 0;
+
+	//Debugging variables
+	int number_of_textures_loaded = 0;
```

Comment:
Maybe give this variables to another class/Struct in order to keep few responsabilities per class

Reply:
Done!